modules/ssl/ssl_engine_vars.c

```diff
@@ -430,10 +430,54 @@ const char *ssl_var_lookup(apr_pool_t *p, server_rec *s,
     return result ? result : "";
 }
 
+#ifdef HAVE_OPENSSL_ECH
+/* Extract ECH status variable from SSL object 'ssl' */
+static const char *ssl_var_lookup_ech_status(apr_pool_t *p, const char *var, SSL *ssl)
+{
+    char *inner_sni = NULL, *outer_sni = NULL;
+    int echrv;
+    const char *result = NULL;
+
+    if (ssl == NULL)
+        return "";
+    echrv = SSL_ech_get1_status(ssl, &inner_sni, &outer_sni);
+    if(strcEQ(var, "ECH_STATUS")) {
+        switch (echrv) {
+        case SSL_ECH_STATUS_NOT_TRIED:
+            result = apr_pstrdup(p, "not attempted");
+            break;
+        case SSL_ECH_STATUS_FAILED:
+            result = apr_pstrdup(p, "tried but failed");
+            break;
+        case SSL_ECH_STATUS_BAD_NAME:
+            result = apr_pstrdup(p, "ECH worked but bad name");
+            break;
+        case SSL_ECH_STATUS_SUCCESS:
+            result = apr_pstrdup(p, "success");
+            break;
+        default:
+            result = apr_pstrdup(p, "error getting ECH status");
+        }
+    }
+    else if (echrv == SSL_ECH_STATUS_SUCCESS) {
+        if(strcEQ(var, "ECH_INNER_SNI")) {
+            result = apr_pstrdup(p, inner_sni);
+        }
+        if(strcEQ(var, "ECH_OUTER_SNI")) {
+            result = apr_pstrdup(p, outer_sni);
+        }
+    }
+    OPENSSL_free(inner_sni);
+    OPENSSL_free(outer_sni);
+    return result ? result : "";
```

Comment:
If nothing was found leave it at `NULL`.
```suggestion
    return result;
```

Reply:
sure